travel-agency-service/src/test/java/cz/muni/fi/pa165/bluebat/service/facade/CustomerFacadeImplTest.java

```diff
@@ -0,0 +1,168 @@
+    }
+
+    @Test
+    public void createCustomer_valid() {
+        Customer customer = getDefaultCustomer();
+        CustomerCreateDTO customerCreateDTO =getDefaultCustomerCreateDTO();
+
+        customerFacade.createCustomer(customerCreateDTO);
+
+        verify(customerService, times(1)).addCustomer(customer);
+    }
+
+    @Test
+    public void updateCustomer_valid() {
+        Long id = 1L;
+        Customer customer = getDefaultCustomer();
+        customer.setId(id);
+        CustomerDTO customerDTO = getDefaultCustomerDTO();
+        customerDTO.setId(id);
+
+        customerFacade.updateCustomer(customerDTO);
+
+        verify(customerService, times(1)).updateCustomer(customer);
+    }
+
+    @Test
+    public void deleteCustomer_valid() {
+        Long id = 1L;
+        Customer customer = getDefaultCustomer();
+        customer.setId(id);
+        when(customerService.findCustomerById(id)).thenReturn(customer);
+
+        customerFacade.deleteCustomer(id);
+
+        verify(customerService, times(1)).deleteCustomer(customer);
+    }
+
+    @Test
+    public void getCustomerById_valid() {
+        Long id = 1L;
+        Customer customer = getDefaultCustomer();
+        customer.setId(id);
+        CustomerDTO customerDTO = getDefaultCustomerDTO();
+        customerDTO.setId(id);
+        when(customerService.findCustomerById(id)).thenReturn(customer);
+
+        CustomerDTO result = customerFacade.getCustomerById(id);
+
+        verify(customerService, times(1)).findCustomerById(id);
+        Assert.assertEquals(result.getId(), customerDTO.getId());
+        Assert.assertEquals(result.getAddress(), customerDTO.getAddress());
+        Assert.assertEquals(result.getName(), customerDTO.getName());
+        Assert.assertEquals(result.getBirthday(), customerDTO.getBirthday());
+        Assert.assertEquals(result.getEmail(), customerDTO.getEmail());
+        Assert.assertEquals(result.getSurname(), customerDTO.getSurname());
+        Assert.assertEquals(result.getPhoneNumber(), customerDTO.getPhoneNumber());
+    }
+    @Test
+    public void getCustomerById_invalidId() {
+        Long id = 1L;
+        when(customerService.findCustomerById(any())).thenReturn(null);
+
+        customerFacade.getCustomerById(1L);
+
+        verify(customerService, times(1)).findCustomerById(id);
+    }
+
+    @Test
+    public void getAllCustomers_valid() {
+        Long idFirst= 1L;
+        Long idSecond= 2L;
+        Customer first = getDefaultCustomer();
+        Customer second=getDefaultCustomer();
+        first.setId(idFirst);
+        second.setId(idSecond);
+        List<Customer>customers = new ArrayList<>();
+        customers.add(first);
+        customers.add(second);
+
+        when(customerService.findAllCustomers()).thenReturn(customers);
+
+        List<CustomerDTO> result = customerFacade.getAllCustomers();
+
+        verify(customerService, times(1)).findAllCustomers();
```

Comment:
Tady se určitě dá ještě ověřit, že ten result je správný. Jinak nemá smysl ho tam mít